icu4c/source/common/brkeng.cpp

```diff
@@ -299,6 +301,86 @@ ICULanguageBreakFactory::loadDictionaryMatcherFor(UScriptCode script) {
     return nullptr;
 }
 
+
+URegistryKey ICULanguageBreakFactory::addExternalEngine(
+    ExternalBreakEngine* engine, UErrorCode& status) {
+    if (U_FAILURE(status)) {
+        return nullptr;
+    }
+    ensureEngines(status);
+    if (U_FAILURE(status)) {
+        return nullptr;
+    }
+    ExternalLanguageBreakEngine* wrapper = new ExternalLanguageBreakEngine(engine);
+    fEngines->push(wrapper, status);
+    return (URegistryKey)wrapper;
+}
+
+UBool ICULanguageBreakFactory::removeExternalEngine(
+    URegistryKey key, UErrorCode& status) {
+    if (U_FAILURE(status)) {
+        return false;
+    }
+    ensureEngines(status);
+    if (U_FAILURE(status)) {
+        return false;
+    }
+    int32_t found = fEngines->search((ExternalLanguageBreakEngine*)key);
+    if (found < 0) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+        return false;
+    }
+    fEngines->removeElementAt(found);
+    return true;
+}
+
+ExternalLanguageBreakEngine::ExternalLanguageBreakEngine(
+    ExternalBreakEngine* engine) : delegate(engine) {
+}
+
+ExternalLanguageBreakEngine::~ExternalLanguageBreakEngine() {
+    delete delegate;
+}
+
+UBool ExternalLanguageBreakEngine::handles(UChar32 c, const char* locale) const {
+    return delegate->isFor(c, locale);
+}
+
+int32_t ExternalLanguageBreakEngine::findBreaks(
+    UText *text, int32_t, int32_t endPos, UVector32 &foundBreaks,
```

Comment:
What's the unnamed `int32_t` parameter here?  From context, it looks like it'd be `startPos`.  If the caller passes you a `startPos`, why are you not using it?  Or is it something else?